src/TLuaInterpreter.cpp

```diff
@@ -15525,18 +15505,11 @@ void TLuaInterpreter::parseMSSP(const QString& string_data)
                 return;
             }
 
-            QString msspVAR;
-            QString msspVAL;
+            QString msspVAR = payloadList[0];
+            QString msspVAL = payloadList[1];
 
-            for (int j = 0; j < payloadList.size(); j++) {
-                if (j < 1) {
-                    msspVAR = payloadList[j];
-                    lua_pushstring(L, msspVAR.toUtf8().constData());
-                } else {
-                    msspVAL = payloadList[j];
-                    lua_pushlstring(L, msspVAL.toUtf8().constData(), msspVAL.toUtf8().length());
-                }
-            }
+            lua_pushstring(L, msspVAR.toUtf8().constData());
```

Comment:
This looks suspicious - you are only handling a single variable name/value pair here whereas the previous code look to do something *different*?

:warning: I am experiencing web-browser/GitHub issues that mean I *may* not be seeing all the contiguous lines of code - in this chunk I can see new lines 15505 to 15515 so if there is extra code related to handling extra values in new lines 15495 to 15504 and 15533 to 15530 (both ranges inclusive) then they are being concealed from me.

Reply:
No I am not doing anything different. Look at the surrounding code which Github helpfully hid from you. Previously there were two nested loops here. The outer loop, which I kept, iterates over all key/value pairs (separated by a 0x01 byte). The code then splits each pair by key/value (hopefully), which are separated by a 0x02 byte.

It then checks whether the key/value pair indeed has two parts, then loops on that, only to assign the first part to `msspVAR` and the second one to `msspVAL`. This inner loop is completely superfluous, we can access the two parts directly. I don't know what @mpconley thought when he wrote that last year, but I can relate, as I've been guilty of coding Rube Goldberg machines too. ;-)

I verified that the result does not change: Lua's `mssp` global contains the same data as before.